File: backend/src/app/schemas/skills.py
from src.app.schemas.utils import CamelCaseModel


class SkillBase(CamelCaseModel):
    """Schema of a skill"""
    name: str
    description: str | None = None


class Skill(CamelCaseModel):
    """Schema of a created skill"""
    skill_id: int
    name: str
    description: str | None = None

    class Config:
        orm_mode = True


class SkillList(CamelCaseModel):
    """A list of Skills"""
    skills: list[Skill]

    class Config:
        orm_mode = True

File: frontend/src/data/interfaces/skills.ts
/**
 * Data about a skill
 */
export interface Skill {
    /** The id of the skill */
    skillId: number;

    /** The name of the skill */
    name: string;
}

/**
 * To create a skill
 */
export interface CreateSkill {
    /** The name of the skill */
    name: string;
}

/**
 * A list of skills
 */
export interface Skills {
    /** The list of skills */
    skills: Skill[];
}
